drivers/sdhc/sdhci_common.c

```diff
@@ -474,6 +545,45 @@ static void sdhci_prepare_data(struct sdhci_common *sdhci_ctx, struct sdhc_comma
 	sys_write16(transfer_mode, sdhci_ctx->reg_base + SDHCI_TRANSFER_MODE);
 }
 
+static int sdhci_dma_check(struct sdhci_common *sdhci_ctx, struct sdhc_data *data)
+{
+	uint32_t trans_bytes;
+
+	if (!sdhci_ctx->f_use_dma) {
+		return 0;
+	}
+
+#if defined(CONFIG_64BIT)
+	/* Only 32bit DMA supported */
```

Comment:
Why can't you use the buffer register, aka non-DMA mode? Do we really need to panic here?

Reply:
I consider as critical mis-configuration with current implementation, so yes panic or assert